v2/invoke.go

```diff
@@ -73,6 +74,11 @@ func invoke(ctx context.Context, call APICall, settings CallSettings, sp sleeper
 		if settings.Retry == nil {
 			return err
 		}
+		// Never retry permanent certificate errors. (e.x. if ca-certificates
+		// are not installed)
+		if strings.Contains(err.Error(), "x509: certificate signed by unknown authority") {
```

Comment:
Yes, unfortunatey: [wrapped in a string](https://github.com/grpc/grpc-go/blob/master/picker_wrapper.go#L146).

re: other certs error - I was thinking of just keeping it as is until we find the need to consider other errors. I'm not a fan of adding string parsing egregiously. Would love to hear your thoughts, though?

Reply:
A different take: if a certificate is signed by an unknown authority, shouldn't this be a more "permanent" error in gRPC?

Reply:
What does "permanent error" mean? gRPC's current behavior [1]is to fail RPCs when no connection exists, whilst in the background continuing to execute connection attempts [2], if that answers the question heh.

1: This is FailFast=true (default) behavior.
2: This happens in the background instead of blocking during Dial because we do not specify WithBlock.

Reply:
Let me see if I understand this right. Certificate signed by unknown authority is considered temporary by gRPC because it will try to connect in the background, but it's "permanent" failure to us?

This seems strange that we disagree on the permanent-ness of the failure. Am I making any sense?

Reply:
I think the terms permanent and transient kind of confuse the issue. Suffice it to say that gRPC is behaving the way it is because it's configured with `block=false` (Dial doesn't block until a connection is established), `failfast=true` (RPCs immediately fail if there's no connection, instead of waiting for one).

So, given these settings, our Dial "succeeds" but our RPCs fail with,

```
rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection
error: connection error: desc = "transport: authentication handshake failed: x509: certificate
signed by unknown authority"
```

Back to this CL: this CL just makes us treat these failing RPCs as a non-retryable error that the user should be aware of. Many status=Unavailable are transient, though, hence why this CL is doing string parsing instead of simply making Unavailable a non-retried code elsewhere.

Reply:
Done